src/gluonts/model/naive_2/_predictor.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Iterator, Optional
+
+# Third-party imports
+import numpy as np
+import pandas as pd
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset, DataEntry
+from gluonts.model.forecast import SampleForecast, Forecast
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.evaluation import get_seasonality
+import statsmodels.api as sm
+
+
+def seasonality_test(past_ts_data, season_length):
+    """
+    Test the time-series for seasonal patterns by performing a 90% auto-correlation test:
+
+    As described here: https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+    Code based on: https://github.com/Mcompetitions/M4-methods/blob/master/Benchmarks%20and%20Evaluation.R
+    """
+    critical_z_score = 1.645  # corresponds to 90% confidence interval
+    if len(past_ts_data) < 3 * season_length:
+        is_seasonal = False
+    else:
+        # calculate auto-correlation for lags up to season_length
+        auto_correlations = sm.tsa.stattools.acf(
+            past_ts_data, fft=False, nlags=season_length
+        )
+        auto_correlations[1:] = 2 * auto_correlations[1:] ** 2
+        limit = (
+            critical_z_score
+            / np.sqrt(len(past_ts_data))
+            * np.sqrt(np.cumsum(auto_correlations))
+        )
+        is_seasonal = (
+            abs(auto_correlations[season_length]) > limit[season_length]
+        )
+
+    return is_seasonal
+
+
+def naive_02(past_ts_data, forecast_start_time, forecast_length):
```

Comment:
I think this function could get past data, period length (integer), and forecast length, and just return the array with the prediction. The predictor below will take care of mapping the frequency of the data to the period length (eg using the get_seasonality utility function), and wrap/unwrap inputs and outputs of this function. What do you think? I think it would make it a more independent utility.

So for example, in computing OWA you don’t have to take the median of the forecast, but you get the array directly.

Also, if you add type annotations it is clear whether we’re dealing with lists or arrays here.

Reply:
I think even though it might be more independent if we just require `period_length` as an argument, it will reduce the convenience of the method, because we usually only have the frequency at hand. The user would need to know how to convert that to `period_length`. Therefore I think it might be best to give `freq` as an argument, and maybe an optional `period_length` that would take precedence if specified. What do you think?

With the rest I'm 100% with you.

Reply:
> Therefore I think it might be best to give `freq` as an argument, and maybe an optional `period_length` that would take precedence if specified. What do you think?

Yes, that's a good compromise. I'd say both optional, at least one should be specified and not both. If `freq` is specified then some default mapping `freq -> period` is applied, otherwise the provided `period` is used.